### ISSUE TITLE
User auth/management component

### ISSUE THREAD

**Issue by GZ82** [Lightning-AI/lightning#14158]
### Motivation

As a backend developer, I mainly focus on algorithms and backend logic.
When to release my app as a product, normally I am lazy to learn knowledge about user management or user authentication function which however is highly desired in many cases. Although I can build this kind of components from scratch but it will be nice if Lightning can implement it as a routine of the workflow.

### Pitch

To summarise the communications with community members. 
The simple version of my request is
" 
I build an app. how can i get auth for the users of my app so i don’t have to code this myself
--- William Falcon
"
or something similar to Cognito in AWS.
The detailed version:
Inbuilt user authentication/authorization mechanism that tracks what a user does in each session.  --- Nash
to realise this, we need the following features ( --- proposed by Sherin):
* A Sign-in/Sign-up page
* An option to add/remove/list users
on the server side
* An object method that other components can then use for validating the session 
* The object method can also manage the session for each user, e.g., allocate computation resources, handling temporary data generated during the session, send email to the user to report status etc. (this probably is too much could be optional :)

### Alternatives

Probably can take advantage of AWS Cognito to achieve those functions mentioned as above.

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

**Comment by tchaton** [comment]
Hey @GZ82 ,

This is definitely on our roadmap and we are planning to provide a built-in Auth system directly embedded within the framework. 

We will keep you updated when we have something for you to try out.

Best,
T.C
